dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/icon/IconTest.java

```diff
@@ -87,12 +88,30 @@ protected void initTest() throws IOException {
   void shouldGetAllIconsWhenRequested() {
     Map<String, DefaultIcon> defaultIconMap = getAllDefaultIcons();
 
+    IconCriteria iconCriteria = new IconCriteria();
+    iconCriteria.setType(IconType.ALL);
+    iconCriteria.setKeywords(List.of());
+
+    assertEquals(
+        defaultIconMap.size() + 1,
+        iconService.getIcons(iconCriteria).size(),
+        String.format(
+            "Expected to find %d icons, but found %d instead",
+            defaultIconMap.size() + 1, iconService.getIcons(iconCriteria).size()));
```

Comment:
Since this is an integration test it would be hard to debug on github if that assertion fails. I would print the list in addition, maybe like

```suggestion
        String.format(
            "Expected to find %d icons, but found %d instead: %s",
            defaultIconMap.size() + 1, iconService.getIcons(iconCriteria).size(), iconService.getIcons(iconCriteria)));
```